lib/ui/progress.coffee

```diff
@@ -0,0 +1,43 @@
+{formatTimePeriod} = require '../misc'
+
+module.exports =
+  progs: {}
+
+  add: (p) ->
+    pr = @ink.progress.create p
+    pr.register()
+    pr.t0 = Date.now()
+    pr.showTime = true
+    @progs[pr.id] = pr
+
+  progress: (p, prog) ->
+    pr = @progs[p.id]
+    return unless pr?
+    pr.setProgress prog
+    if pr.showTime then @rightText pr
+
+  message:  (p, m) -> @progs[p.id]?.setMessage  m
+
+  leftText: (p, m) -> @progs[p.id]?.setLeftText m
```

Comment:
Fair enough. Grouping sounds like a good idea, should there ever be some other language plugin that uses our porgress bars ;)